Description: Properly handle exceptions about hg extra fields.
Origin: commit, revision id: jelmer@jelmer.uk-20190113141635-f7p75zxtisiidha7
Author: Jelmer VernooÄ³ <jelmer@jelmer.uk>
Last-Update: 2019-01-13
Applied-Upstream: no
X-Bzr-Revision-Id: jelmer@jelmer.uk-20190113141635-f7p75zxtisiidha7

=== modified file 'breezy/git/errors.py'
--- old/breezy/git/errors.py	2018-11-16 11:37:47 +0000
+++ new/breezy/git/errors.py	2019-01-13 14:16:35 +0000
@@ -78,4 +78,4 @@
     def __init__(self, object, fields):
         brz_errors.BzrError.__init__(self)
         self.object = object
-        self.fields = ",".join(fields)
+        self.fields = b",".join(fields)

=== modified file 'breezy/git/tests/test_mapping.py'
--- old/breezy/git/tests/test_mapping.py	2018-11-11 04:08:32 +0000
+++ new/breezy/git/tests/test_mapping.py	2019-01-13 14:16:35 +0000
@@ -34,7 +34,10 @@
     )
 
 from .. import tests
-from ..errors import UnknownCommitExtra
+from ..errors import (
+    UnknownCommitExtra,
+    UnknownMercurialCommitExtra,
+    )
 from ..mapping import (
     BzrGitMappingv1,
     escape_file_id,
@@ -217,6 +220,22 @@
         self.assertEqual(
             rev.properties[u'git-mergetag-0'], tag.as_raw_string())
 
+    def test_unknown_hg_fields(self):
+        c = Commit()
+        c.tree = b"cc9462f7f8263ef5adfbeff2fb936bb36b504cba"
+        c.message = b"Some message"
+        c.committer = b"Committer"
+        c.commit_time = 4
+        c.author_time = 5
+        c.commit_timezone = 60 * 5
+        c.author_timezone = 60 * 3
+        c.author = b"Author"
+        c._extra = [(b"HG:extra", b"bla:Foo")]
+        mapping = BzrGitMappingv1()
+        self.assertRaises(
+            UnknownMercurialCommitExtra,
+            mapping.import_commit, c, mapping.revision_id_foreign_to_bzr)
+
 
 class RoundtripRevisionsFromBazaar(tests.TestCase):
 

=== modified file 'breezy/git/mapping.py'
--- old/breezy/git/mapping.py	2018-11-16 23:21:31 +0000
+++ new/breezy/git/mapping.py	2019-01-13 18:59:26 +0000
@@ -452,7 +452,7 @@
             elif k == HG_EXTRA:
                 hgk, hgv = v.split(b':', 1)
                 if hgk not in (HG_EXTRA_AMEND_SOURCE, ):
-                    raise UnknownMercurialCommitExtra(commit, hgk)
+                    raise UnknownMercurialCommitExtra(commit, [hgk])
                 extra_lines.append(k + b' ' + v + b'\n')
             else:
                 unknown_extra_fields.append(k)

